### PR TITLE
Automatically translate the default maintenance template

### PR DESCRIPTION
**Pull request by Toflar**
Here's a solution that covers the 99% case. Most people won't adjust the maintenance template and so we can use JS to translate it according to the user's browser language if we have translations available.

So for German, this now looks like this automatically:

Bildschirmfoto 2023-11-07 um 17 24 20

We might improve this even more in the future by using the upcoming Symfony translation support for JS but this is still experimental and I don't know if this is really a good fit anyway. The reason for that is that it would tie the maintenance mode to require a running system again.
I know that we are already sort of relying on the fact that the container still works as we're using Twig when using `contao:maintenance enable`. So I guess this is kind of meant to work like "in case the System does work, we can use your template, if not, then you have to copy something to `var/maintenance.html` yourself".

So Twig should be fine. But using dynamic translations would probably be a bit too much.

PS: Translating the thing in the footer doesn't work because of placeholders. So this here is really KISS.

### CODE REVIEW

File: core-bundle/templates/Error/service_unavailable.html.twig [hunk 1]
```diff
@@ -4,5 +4,36 @@
...
+                }
+                if (data[language][key] === undefined) {
+                    return null;
+                }
+                return data[language][key];
+            }
+            d.querySelectorAll('[data-trans]').forEach(el => {
+                let trans = translate(el.dataset.trans);
+                if (trans) {
+                    el.innerHTML = trans;
```
Comment (ausi) [inline core-bundle/templates/Error/service_unavailable.html.twig]:
```suggestion
                    el.textContent = trans;
```

If I’m right, the messages get encoded in the twig template, so we should not allow HTML here I think.

File: manager-bundle/src/Command/MaintenanceModeCommand.php [hunk 2]
```diff
@@ -119,4 +124,21 @@ private function outputResult(InputInterface $input, OutputInterface $output, bo
             $io->info($message);
         }
     }
+
+    private function generateDefaultLabels(): array
+    {
+        $labels = [];
+
+        foreach (array_keys($this->locales->getEnabledLocales()) as $locale) {
```
Comment (ausi) [inline manager-bundle/src/Command/MaintenanceModeCommand.php]:
```suggestion
        foreach ($this->locales->getEnabledLocaleIds() as $locale) {
```

**Review by ausi — CHANGES_REQUESTED** [review]
LGTM apart from the two small changes 👍

**Review by ausi — DISMISSED** [review]

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @Toflar.

**Pull request merged**